src/app/pages/search-character/search-character.component.ts

```diff
@@ -32,6 +38,13 @@ export class SearchCharacterComponent {
     this.getCharacters();
   }
 
+  goToResults(): void {
+    this.filtersService.addFilter('name', this.currentInput);
+    this.filtersService.addFilter('species', this.currentSpecie);
+    this.filtersService.addFilter('gender', this.currentGender);
```

Comment:
- check if this.filtersService.addFilter  could receive an object with the values ​​so as not to be executing so many processes

Reply:
It could be possible. However, I prefer to leave the logic simple on the service rather than creating an object iterator.
Setting the filters by one is suitable for any component. 